elastic/kibana
# [Alerting] Added retry logic in the alerting task run during update rule that fails with Saved object not found error.

Resolves #100764
## Summary

Current PR adding a retry logic for the rule task runner and action tasks in case if SO 404 Not Found errors happen to avoid accidentally removing tasks for the false 'not found' objects.

How to test the logic:
1. start local ES with `yarn es snapshot --ssl -E path.data=/../es-data`
2. start Kibana `yarn start --ssl`
3. Create a rule which firing often with 5s interval (use `es-apm-sys-sim -r 40 15 es-apm-sys-sim https://elastic:changeme@localhost:9200` to simulate the data) with some actions (like Server Log)
4. Check if the rule task is successfully running in the logs and UI
5. Stop ES server for a second and start it with the same command `yarn es snapshot --ssl -E path.data=/../es-data`
6. Wait till it will start and Kibana get back to the normal health
7. Observer that previously create rule continue running and actions still executed successfully 

No need to Disable/Enable rule to make it works.
